zebra-network/src/isolated.rs

```diff
@@ -101,7 +130,35 @@ pub fn connect_isolated_tcp_direct(
     addr: SocketAddr,
     user_agent: String,
 ) -> impl Future<Output = Result<BoxService<Request, Response, BoxError>, BoxError>> {
+    let nil_inbound_service =
+        tower::service_fn(|_req| async move { Ok::<Response, BoxError>(Response::Nil) });
+
+    connect_isolated_tcp_direct_with_inbound(network, addr, user_agent, nil_inbound_service)
+}
+
+/// Creates an isolated Zcash peer connection using the provided data stream.
+/// This function is for testing purposes only.
+///
+/// See [`connect_isolated_with_inbound`] and [`connect_isolated_tcp_direct`] for details.
+///
+/// # Privacy
+///
+/// This function can make the isolated connection send different responses to peers,
+/// which makes it stand out from other isolated connections from other peers.
+pub fn connect_isolated_tcp_direct_with_inbound<InboundService>(
```

Comment:
Question: Would it make sense to make this function be only available for test builds?

```suggestion
#[cfg(any(test, feature = "proptest-impl"))]
pub fn connect_isolated_tcp_direct_with_inbound<InboundService>(
```

Reply:
This function already has this exact restriction when *exported* in `lib.rs`.